src/devices/GoPiGo3/GoPiGo3.cs

```diff
@@ -0,0 +1,887 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Iot.Device.GoPiGo3.Models;
+using System;
+using System.Device.Spi;
+using System.Device.Spi.Drivers;
+using System.Diagnostics;
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+
+namespace Iot.Device.GoPiGo3
+{
+    /// <summary>
+    /// A GoPiGo3 class for a GoPiGo3 Dexter Industries robot kit
+    /// </summary>
+    public class GoPiGo : IDisposable
+    {
+        private const int DefaultMotorGearRatio = 120;
+        private const int DefaultEncoderTicksPerRotation = 6;
+        private const int GrooveI2cLengthLimit = 32;
+        private const string ManufacturerName = "Dexter Industries";
+        private const string BoardName = "GoPiGo3";
+        private static SpiDevice _spiDevice = null;
+        private const byte CorrectData = 0xA5;
```

Comment:
>CorrectData [](start = 27, length = 11)

Can we rename these constants to explain a bit better what do they mean or are used for? also, apart from renaming can we add a comment above them with more verbose description?

Reply:
will do and Added a comment to explain both constants